### PR TITLE
Add an example of playback using gstreamer.

### PR DESCRIPTION
**Pull request by andrew-otiv**
Add a very limited example of gstreamer output in a slint window.  Has assets for play/pause buttons from the ffmpeg demo, but no logic for them yet.

### CODE REVIEW

File: examples/gstreamer_slint_demo/video.rs
Comment (tronical) [inline examples/gstreamer_slint_demo/video.rs:1]: I think this file may not be needed? :)

File: Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ members = [
     'examples/opengl_underlay',
     'examples/opengl_texture',
     'examples/ffmpeg',
+    'examples/gstreamer_slint_demo',
```
Comment (tronical) [inline Cargo.toml:21]: I suggest to just call it `gstreamer`

File: examples/gstreamer_slint_demo/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+[dependencies]
+slint = { path = "../../api/rs/slint" }
+anyhow = { version = "1.0" }
+futures = { version = "0.3.28" }
+
+gst = {package="gstreamer", version="0.21.3"}
+gst-audio = {package="gstreamer-audio",version="0.21.3"}
+gst-video = {package="gstreamer-video",version="0.21.2"}
+gst-app = {package="gstreamer-app",version="0.21.2"}
```
Comment (tronical) [inline examples/gstreamer_slint_demo/Cargo.toml:25]: Just some nitpicking ;)

```suggestion
gst = {package = "gstreamer", version = "0.21.3"}
gst-audio = {package = "gstreamer-audio", version = "0.21.3" }
gst-video = {packag e= "gstreamer-video", version = "0.21.2" }
gst-app = {package = "gstreamer-app", version = "0.21.2" }
```

**Review by tronical — COMMENTED** [review]
Nice idea - looks overall good to me. Thanks! I've added a few suggestions inline, and I suggest to also run `cargo fmt` to match the code formatting in one go.

If you want to play with it, CI coverage would also be nice. Take a look at ci.yaml and search for ffmpeg: I think you can duplicate that block and adjust it for streamer as well. Then extend the `--exclude ffmpeg` bits in the same file to also exclude gstreamer, so that it's only tested on Linux.

File: examples/gstreamer_slint_demo/video.rs
Comment (andrew-otiv) [reply examples/gstreamer_slint_demo/video.rs:1]: oeps! will delete.

File: examples/gstreamer_slint_demo/Cargo.toml (hunk 2 above)
Comment (tronical) [reply examples/gstreamer_slint_demo/Cargo.toml:25]: Ooops, thanks  - fixed :)

File: Cargo.toml (hunk 1 above)
Comment (andrew-otiv) [reply Cargo.toml:21]: Tools were having issues due to the name collision with the gstreamer crate. I was working in vscode. I think I just had the example loaded in the IDE. Helps humans too IMO. That probably affects the ffmpeg example code too, but may depend on IDE and IDE config. Still want it "gstreamer"?

Comment (tronical) [reply Cargo.toml:21]: Fair point. How about `gstreamer-player`?

File: examples/gstreamer_slint_demo/video.rs
Comment (andrew-otiv) [reply examples/gstreamer_slint_demo/video.rs:1]: RE The CICD coverage, I read your CI config, and I couldn't find where examples are actually getting built and run... maybe via the tests? For example, I can't tell if your --exclude ffmpeg was actually doing anything, but if it's a crate name or a pattern I went ahead and added the example name.

I'm not very well versed in gilab CI.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -53,6 +53,12 @@ jobs:
     - name: Install ffmpeg and alsa (Linux)
       if: runner.os == 'Linux'
       run: sudo apt-get install clang libavcodec-dev libavformat-dev libavutil-dev libavfilter-dev libavdevice-dev libasound2-dev pkg-config
+    - name: Install gstreamer (Linux)
+      if: runner.os == 'Linux'
+      run: sudo apt-get install libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev \
+        gstreamer1.0-plugins-base gstreamer1.0-plugins-good
```
Comment (tronical) [inline .github/workflows/ci.yaml:59]:
```suggestion
      run: sudo apt-get install libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev gstreamer1.0-plugins-base gstreamer1.0-plugins-good
```

File: examples/gstreamer_slint_demo/video.rs
Comment (tronical) [reply examples/gstreamer_slint_demo/video.rs:1]: No worries, if you'd like I can try to fix up the CI bits and push them into this PR.

Comment (andrew-otiv) [reply examples/gstreamer_slint_demo/video.rs:1]: Yes please, at your leisure since it's just an example, and not a particularly complete one at that.

Comment (andrew-otiv) [reply examples/gstreamer_slint_demo/video.rs:1]: Actually, please hold off on it... our company is transitioning CICD soon, so we may not have github credits for the CI runs on this branch.

File: Cargo.toml (hunk 1 above)
Comment (andrew-otiv) [reply Cargo.toml:21]: I'm OK with the name change, but if we call it a player, I should at least put a warning about the buttons not working in the example README.

Comment (tronical) [reply Cargo.toml:21]: Sounds good to me.

File: examples/gstreamer_slint_demo/video.rs
Comment (andrew-otiv) [reply examples/gstreamer_slint_demo/video.rs:1]: Didn't realise CICD runs happen in the target repo's organization.
I fixed up a few minor things, build is green now!

**Review by tronical — APPROVED** [review]
Thanks for the example :)

**Pull request merged**